api/src/main/java/ca/bc/gov/educ/api/batchgraduation/reader/RegenerateCertificatePartitioner.java

```diff
@@ -89,6 +90,15 @@ private List<StudentCredentialDistribution> getStudentsForUserReqRun(Certificate
         }
     }
 
+    private UUID getStudentIDByPen(String pen) {
+        try {
```

Comment:
Need to log error if there is an exception encountered calling getStudentIDByPen(). For example, 5xx or 4xx errors etc.

Reply:
getStudentIDByPen() is retried thru resilience4j.   Do you want to change it to the generic get method?

Reply:
I think that might be better. We might as well clean as we code =)

Reply:
Resilience4j.retry has been replaced with the generic methods.